lib/controllers/BookmarkController.dart

```diff
@@ -1,33 +1,82 @@
+import 'package:bookmarks/abstractions/Controller.dart';
 import 'package:bookmarks/models/Bookmark.dart';
+import 'package:bookmarks/models/Folder.dart';
 import 'package:bookmarks/models/User.dart';
 import 'package:bookmarks/services/BookmarkService.dart';
-import 'package:bookmarks/widgets/HomeWidget.dart';
+import 'package:bookmarks/services/FolderService.dart';
 import 'package:bookmarks/views/BookmarkListView.dart';
-import 'package:bookmarks/abstractions/Controller.dart';
+import 'package:bookmarks/widgets/HomeWidget.dart';
 import 'package:flutter/cupertino.dart';
 import 'package:url_launcher/url_launcher.dart';
 
 class BookmarkController extends Controller<HomeWidget> {
   int _counter = 0;
 
+  int _folderId = -1;
```

Comment:
Are folder ids guaranteed to be positive?

Reply:
Yes, `-1` is the magic id of the root folder for every user.